### PR TITLE
PARQUET-353: Recycle compressors in parquet write path

### PR DESCRIPTION
**Pull request by nitin2goyal**

### CODE REVIEW

File: parquet-hadoop/src/main/java/parquet/hadoop/ParquetFileWriter.java [hunk 1]
```diff
@@ -143,7 +143,9 @@ private final STATE error() throws IOException {
 
   private STATE state = STATE.NOT_STARTED;
 
-  /**
+  private final CodecFactory codecFactory;
+
+/**
```
Comment (liancheng) [inline parquet-hadoop/src/main/java/parquet/hadoop/ParquetFileWriter.java]: Nit: Indentation is off.

File: parquet-hadoop/src/main/java/parquet/hadoop/ParquetFileWriter.java [hunk 2]
```diff
@@ -444,6 +448,13 @@ static ParquetMetadata mergeFooters(Path root, List<Footer> footers) {
   public long getPos() throws IOException {
     return out.getPos();
   }
+  
+  /**
+   * @return the code factory which should be used to get compressors
```
Comment (liancheng) [inline parquet-hadoop/src/main/java/parquet/hadoop/ParquetFileWriter.java]: "the code ..." => "The codec ..."

**Comment by rschmitt** [comment]
Is this ready to merge?

**Comment by liancheng** [comment]
Hey @nitin2goyal, I just realized that this PR is opened against the wrong repository. parquet-mr has been moved to https://github.com/apache/parquet-mr/. Sorry that I didn't realize this earlier.

**Comment by nitin2goyal** [comment]
Opened fresh PR under correct Repo

https://github.com/apache/parquet-mr/pull/282

On Mon, Oct 19, 2015 at 7:53 AM, Cheng Lian <notifications@github.com>
wrote:

> Hey @nitin2goyal <https://github.com/nitin2goyal>, I just realized that
> this PR is opened against the wrong repository. parquet-mr has been moved
> to https://github.com/apache/parquet-mr/. Sorry that I didn't realize
> this earlier.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Parquet/parquet-mr/pull/445#issuecomment-149079540>.
>



-- 
Regards
Nitin Goyal

**Comment by nitin2goyal** [comment]
Closing in favor of https://github.com/apache/parquet-mr/pull/282

**Pull request closed without merging**